todo_app/app.py

```diff
@@ -59,28 +75,34 @@ def index():
     @login_required
     @ReadWriteNeeded
     def add_item(): 
-        to_do_list_local.add_card(request.form.get('title'))
+        title = request.form.get('title')
+        to_do_list_local.add_card(title)
+        app.logger.info(f"User {current_user.id} added to do item {title}")
```

Comment:
Minor, but worth noting that this assumes that your current_user _has_ an id - but when you switch off your login behaviour then anonymous users cause this to crash! We can fix this by setting our own AnonymousUser class, or by replacing this with a function that is careful:
```python
def get_id():
    if os.getenv('LOGIN_DISABLED') == 'True':
        return "Anonymous"
    else:
        return current_user.id
```